include/networkit/graph/GraphTools.hpp

```diff
@@ -418,6 +419,63 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Randomizes the weights of the given graph. The weights are uniformly distributed in
+ * the range [0, 1] by default, unless a different distribution is provided. However it
+ * is only strictly in-place for already weighted graphs. For unweighted graphs a copy is
+ * created before randomizing weights.
+ *
+ * @param   G           Directed input graph
+ * @param   distr       Random distribution
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(Graph &G, Distribution distr = std::uniform_real_distribution<edgeweight>{
+                                    0, std::nexttoward(1.0, 2.0)}) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+#pragma omp parallel
+    {
+        std::mt19937 gen;
+        auto baseSeed = Aux::Random::integer();
+
+#pragma omp for schedule(static)
+        for (omp_index u = 0; u < G.upperNodeIdBound(); u++) {
```

Comment:
Why not `G.parallelForNodes()`? This would also fail if the graph has deleted nodes.

Reply:
because it wouldn't be deterministic. Also, it doesn't fail with deleted nodes because `outEdges` of a node is not deleted when a node is removed but simply has its edges removed.

Reply:
Alright, then I only a minor recommendation:
```suggestion
#pragma omp for schedule(static)
        for (omp_index u = 0; u < G.upperNodeIdBound(); ++u) {
```